rest/src/main/java/org/jboss/pnc/rest/provider/ProductMilestoneProvider.java

```diff
@@ -91,4 +122,118 @@ public void removeDistributedArtifact(Integer milestoneId, Integer artifactId) t
         repository.save(milestone);
     }
 
+    @Override
+    public void update(Integer id, ProductMilestoneRest restEntity) throws ValidationException {
+        restEntity.setId(id);
+        validateBeforeUpdating(id, restEntity);
+        ProductMilestone milestone = toDBModel().apply(restEntity);
+
+        if (restEntity.getEndDate() != null && noReleaseInProgress(milestone)) {
+            ProductMilestoneRelease release = startRelease(milestone);
+            releaseRepository.save(release);
```

Comment:
Do we need to save the entity before the completion? Can we keep the DB immutable and keep the run-time state in-memory.